framework/widgets/Menu.php

```diff
@@ -210,9 +210,12 @@ protected function renderItems($items)
             }
             if (!empty($class)) {
                 if (empty($options['class'])) {
-                    $options['class'] = implode(' ', $class);
+                    $options['class'] = $class;
                 } else {
-                    $options['class'] .= ' ' . implode(' ', $class);
+                    if (!is_array($options['class'])) {
+                        $options['class'] = [$options['class']];
+                    }
+                    $options['class'] = ArrayHelper::merge($class, $options['class']);
```

Comment:
`$options['class']` is an array so it's not concatenated to `$class` but is merged with it.